crates/iota-framework/packages/iota-system/sources/validator.move

```diff
@@ -608,6 +611,32 @@ module iota_system::validator {
         }
     }
 
+
+    public(package) fun smaller_than(self: &ValidatorV1, other: &ValidatorV1): bool{
+        if (self.total_stake() != other.total_stake()) {
+            return self.total_stake() < other.total_stake()
+        };
+
+        let self_pubkey = self.authority_pubkey_bytes();
+        let other_pubkey = other.authority_pubkey_bytes();
+
+        // Compare the two pubkeys lexicographically, assuming equal lengths
+        let len = self_pubkey.length();
+
+        let mut i = 0;
+        while (i < len) {
+            let self_byte = self_pubkey[i];
+            let other_byte = other_pubkey[i];
+
+            if (self_byte != other_byte) {
+                return self_byte < other_byte
+            };
+            i = i + 1;
+        };
+        
+        false // Should never end up here
```

Comment:
Is there anything that makes sure of that for the authority_pubkey? I only see a check for the iota_address
https://github.com/iotaledger/iota/blob/474ca3a70dc536502f65d96ebaf85d5893158a69/crates/iota-framework/packages/iota-system/sources/validator_set.move#L164-L168
So maybe the ordering here should also use the `iota_address` instead?

Reply:
Here we check that all the fields are unique

https://github.com/iotaledger/iota/blob/62274b80f683ffbe3ad0dce6b3045d1bd94fdaf0/crates/iota-framework/packages/iota-system/sources/validator.move#L562

Reply:
`MUSTFIX: We need to check this when updating metadata as well.` :sweat_smile: